File: frontend/src/types/exam.ts
export interface Exam {
  id: string
  spaceId: string
  spaceCode: string
  spaceName: string
  title: string
  description: string | null
  location: string
  startsAt: string
  endsAt: string
  published: boolean
  createdByUserId: string
  createdAt: string
  updatedAt: string
  canManage: boolean
}

export interface ExamPayload {
  spaceId: string
  title: string
  description: string
  location: string
  startsAt: string
  endsAt: string
  published: boolean
}

File: backend/src/main/java/edusecure/edusecure/entity/exam/Exam.java
package edusecure.edusecure.entity.exam;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.Instant;
import java.util.UUID;

@Entity
@Table(name = "exams")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Exam {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(nullable = false)
    private UUID spaceId;

    @Column(nullable = false, length = 160)
    private String title;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(nullable = false, length = 160)
    private String location;

    @Column(nullable = false)
    private Instant startsAt;

    @Column(nullable = false)
    private Instant endsAt;

    @Column(nullable = false)
    private boolean published;

    @Column(nullable = false)
    private UUID createdByUserId;

    @Column(nullable = false)
    private Instant createdAt;

    @Column(nullable = false)
    private Instant updatedAt;
}